From af173ca74b8ecf40dc6ab00710eca23ec37711cd Mon Sep 17 00:00:00 2001
From: Bala Sai Kosuri <balasaik@xilinx.com>
Date: Thu, 23 Jun 2022 22:06:11 +0530
Subject: [PATCH] Derive src fps from vui_time_scale & vui_num_units_in_tick

Signed-off-by: Bala Sai Kosuri <balasaik@xilinx.com>
---
 gst/videoparsers/gsth265parse.c | 27 ++++++++++++++++++++-------
 1 file changed, 20 insertions(+), 7 deletions(-)

diff --git a/gst/videoparsers/gsth265parse.c b/gst/videoparsers/gsth265parse.c
index f2636b4de..7c12855ec 100644
--- a/gst/videoparsers/gsth265parse.c
+++ b/gst/videoparsers/gsth265parse.c
@@ -1821,14 +1821,27 @@ gst_h265_parse_update_src_caps (GstH265Parse * h265parse, GstCaps * caps)
     }
 
     /* 0/1 is set as the default in the codec parser */
-    if (sps->vui_params.timing_info_present_flag &&
-        !(sps->fps_num == 0 && sps->fps_den == 1)) {
-      if (G_UNLIKELY (h265parse->fps_num != sps->fps_num
-              || h265parse->fps_den != sps->fps_den)) {
+    if (sps->vui_params.timing_info_present_flag) {
+      gint fps_num = 0, fps_den = 1;
+
+      if (!(sps->fps_num == 0 && sps->fps_den == 1)) {
+        fps_num = sps->fps_num;
+        fps_den = sps->fps_den;
+      } else if (!(sps->vui_params.time_scale == 0 &&
+              sps->vui_params.num_units_in_tick == 1)) {
+        fps_num = sps->vui_params.time_scale;
+        fps_den = sps->vui_params.num_units_in_tick;
+
+        if (sps->profile_tier_level.interlaced_source_flag)
+          fps_num /= 2;
+      }
+
+      if (G_UNLIKELY (h265parse->fps_num != fps_num
+              || h265parse->fps_den != fps_den)) {
         GST_INFO_OBJECT (h265parse, "framerate changed %d/%d",
-            sps->fps_num, sps->fps_den);
-        h265parse->fps_num = sps->fps_num;
-        h265parse->fps_den = sps->fps_den;
+            fps_num, fps_den);
+        h265parse->fps_num = fps_num;
+        h265parse->fps_den = fps_den;
         modified = TRUE;
       }
     }
-- 
2.17.1

